server/db/model.py

```diff
@@ -274,14 +282,25 @@ def prospects_bulk_append(self, prospects_list):
         Args:
             prospects_list: list of keyword dict to construct Prospect
         Returns:
-            list: list of id of prospects
+            dict: length of both duplicate prospects as well as new prospects
         """
-        # TODO: handle repeat/exists prospects
-        prospects_objs = Prospect.objects.bulk_create(
-            [Prospect(owner=self._id, **each_p) for each_p in prospects_list], retrieve=False)
-        self.prospects_count += len(prospects_objs)
-        self.save()
-        return prospects_objs
+
+        own_prospect_emails = [prospect.email for prospect in self.prospects]
+
+        new_prospects_list = [
+            prospect_obj for prospect_obj in prospects_list if prospect_obj['email'] not in own_prospect_emails]
+
+        dup_prospects_list = [p for p in prospects_list +
+                              new_prospects_list if p not in prospects_list or p not in new_prospects_list]
```

Comment:
can you make each of these sets for faster access time (instead of checking arrays)

Reply:
Made suggested changes in latest commit